### DISCUSSION TITLE
How to stop PEST execution

### DISCUSSION THREAD

**Discussion by DBCorkran** [pypest/pyemu#487] [Q&A] [answered]
Hi,

I am wondering if there is a way to stop PEST execution (PSTOP or PSTOPST) when you run PEST in pyemu using pyemu.utils.os_utils.run.

**Comment by jtwhite79** [comment, marked as answer]
if you make a file in the directory where pest++ is running called "pest.stp" and put 1 or 2 as the first token on the first line, that will cause it to exit (semi-)gracefully.  

[https://github.com/usgs/pestpp/blob/master/documentation/pestpp_users_manual.md#s9-3-5](https://github.com/usgs/pestpp/blob/master/documentation/pestpp_users_manual.md#s9-3-5)

**Reply by DBCorkran** [reply]
Thank you!
